dist/pathfora.js

```diff
@@ -1601,10 +1637,49 @@
         this.updateObject(widget, defaults);
         this.updateObject(widget, widget.config);
 
-        if (widget.displayConditions.showDelay) {
-          core.registerDelayedWidget(widget);
+        var displayWidget = function(widget) {
+          if (widget.displayConditions.showDelay) {
+            core.registerDelayedWidget(widget);
+          } else {
+            core.initializeWidget(widget);
+          }
+        }
+
+        if (widget.type === "message" && widget.recommend) {
+          if (widget.layout !== "slideout" && widget.layout !== "modal") {
+            throw new Error('Unsupported layout for content recommendation');
+          }
+
+          var def;
+          if (widget.content && widget.content[0]) {
+            if (widget.content[0].default)
+              def = widget.content[0];
+            else
+              throw new Error('Cannot define recommended content unless it is a default');
+          }
+
+          api.recommendContent(accountId, widget.recommend.ql.raw, function(content){
+            var con = [];
+
+            if (content) {
+              con.push({
```

Comment:
Yeah, I think the reason why I did that is because at some point prior to this, any array in the config gets turned into an object, so like if I console log `widget.content` before this it's gonna be `Object {0: Object}` no idea why it does that, but that's why I didn't do `unshift` however, it might make sense to just overwrite the default, as you were saying.